### PR TITLE
feat(user): add "Hide completed sets" checkbox to Completion Progress

### PR DESCRIPTION
**Pull request by wescopeland**
This PR adds a new checkbox to the user page "Completion Progress" component. When checked, completed and mastered games are hidden from the list.

This change was inspired by one of @Xymjak's userscripts. I have credited him on the initial commit.

Most of the lines added in this commit are automated lockfile changes for front-end testing dependencies.

https://user-images.githubusercontent.com/3984985/232251528-510b8462-5b12-4009-954c-c6a5ca1e6c7b.mp4

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,6 +24,9 @@
         "@tailwindcss/forms": "^0.5.2",
         "@tailwindcss/line-clamp": "^0.4.0",
         "@tailwindcss/typography": "^0.5.0",
+        "@testing-library/dom": "^9.2.0",
+        "@testing-library/jest-dom": "^5.16.5",
+        "@testing-library/user-event": "^14.4.3",
```
Comment (wescopeland) [inline package.json:29]: Industry-standard packages for querying, interacting with, and testing headless DOM environments in Jest and Vitest. It wouldn't surprise me if the Vue or React flavors are used in Laravel Livewire settings.

File: resources/js/lazyLoadModuleOnIdFound.ts [hunk 2]
```diff
@@ -5,15 +5,19 @@
...
  * @param {string} options.codeFileName - The name of the file in the dynamic folder to lazy load.
- * @param {string} options.moduleName - The name under which the imported module will be attached to the window object.
+ * @param {string} options.moduleNameToAttachToWindow - The name under which the imported module will be attached to the window object.
  */
-export const lazyLoadModuleOnIdFound = (options: { elementId: string; codeFileName: string; moduleName: string }) => {
-  const { elementId, codeFileName, moduleName } = options;
+export const lazyLoadModuleOnIdFound = (options: {
+  elementId: string;
+  codeFileName: string;
+  moduleNameToAttachToWindow: string;
```
Comment (wescopeland) [inline resources/js/lazyLoadModuleOnIdFound.ts:13]: Renamed this to hopefully make it more clear what this `lazyLoadModuleOnIdFound` function is doing.

File: vite.config.ts [hunk 3]
```diff
@@ -41,7 +41,10 @@ export default defineConfig(({ mode }) => {
         '@': resolve(__dirname, './resources/js'),
       },
     },
-    test: {},
+    test: {
+      environment: 'jsdom',
```
Comment (wescopeland) [inline vite.config.ts:45]: Unexpectedly, this seems to work without an explicit `devDependency` for `jsdom`.

`npm ls jsdom` shows we are pulling in `jsdom` transitively through `vitest`.

**Review by luchaos — APPROVED** [review]

**Review by Jamiras — APPROVED** [review]
I'm going to merge this as-is due to luchaos's approval and my testing.

I think it might be worth adding a minimum threshold before displaying the checkbox though.

Completely new user:
image

User with just a few masteries:
image

**Pull request merged**